### PR TITLE
#33: apim backend sps

### PR DESCRIPTION
**Pull request by aaheiev**
- Add required infrastructure components (APIM RG)
- Add SP's with APIM contributor permissions 
- Save SP's as Kubernetes secrets

### CODE REVIEW

File: infrastructure/services.tf [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  resource_group_name     = azurerm_resource_group.services_rg.name
+  location                = azurerm_resource_group.services_rg.location
+  sku                     = "Standard"
+  admin_enabled           = false
+  anonymous_pull_enabled  = false
+  data_endpoint_enabled   = false
+  zone_redundancy_enabled = false
+}
+
+resource "azurerm_key_vault" "akv" {
```
Comment (lucashuet93) [inline infrastructure/services.tf:24]: Does the application insights instrumentation key get stored as a Key Vault secret?

**Review by lucashuet93 — CHANGES_REQUESTED** [review]
The additions of the AKS related infrastructure are awesome, but I don't want to lose the functionality the repository currently provides. The infrastructure it currently deploys is closely tied to the /apimartifacts contents. For instance, the Application Insights Instrumentation Key is stored as a Key Vault secret via Terraform. The /apimartifacts folder has a named value that point to the Instrumentation Key secret. 

I'd like to add the AKS functionality on top of what currently exists, if possible. It looks like if I deployed the infra, then ran the publisher against said infra as outlined by the README, it would not work properly.

File: infrastructure/datasources.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+data "azuread_client_config" "current" {}
+
+data "azurerm_role_definition" "aks_cluster_admin" {
```
Comment (ross-p-smith) [inline infrastructure/datasources.tf:3]: Can we put a comment on what these magic strings equate to for future developers?

**Comment by ross-p-smith** [comment]
I've checkout this PR and when I run `make infra` I get the following error: 

```
vscode ➜ /workspaces/apiops-vnet-github (33/apim-backend-sps) $ make infra
infra

Initializing the backend...
Initializing modules...
╷
│ Error: Error inspecting states in the "local" backend:
│     Error retrieving keys for Storage Account "apimmvetfstate": storage.AccountsClient#ListKeys: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceGroupNotFound" Message="Resource group 'infra' could not be found."
│ 
│ Prior to changing backends, Terraform inspects the source and destination
│ states to determine what kind of migration steps need to be taken, if any.
│ Terraform failed to load the states. The data in both the source and the
│ destination remain unmodified. Please resolve the above error and try again.
```

**Pull request closed without merging**